platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/MetricConfig.java

```diff
@@ -33,20 +33,7 @@
  * @param <T> the {@code Class} for which the configuration is
  */
 @SuppressWarnings("removal")
-public abstract sealed class MetricConfig<T extends Metric, C extends MetricConfig<T, C>>
-        permits Counter.Config,
```

Comment:
As a next step I want to move MetricConfig to a metrics-api module but some of the concrete implementations won't be moved. In addition I do not understand why it is sealed since you should be able to create custom metrics in future.